src/dataToEsm.ts

```diff
@@ -3,6 +3,10 @@ import { DataToEsm } from './pluginutils';
 
 export type Indent = string | null | undefined;
 
+function stringify (obj :any) :string {
+	return (JSON.stringify(obj) || 'undefined').replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029');
```

Comment:
What is the fallback `undefined` good for? If I remove it, no test turns red and I could not come up with a situation where `JSON.stringify` would return a falsy value.

Reply:
Maybe the regular expressions could be combined? Then it would be easy to add more characters to be escaped:

```js
JSON.stringify(obj).replace(/[\u2028\u2029]/g, char => `\\u${char.charCodeAt(0).toString(16))}`;
```

Reply:
Ah, I see `JSON.stringify(undefined)` returns `undefined`. So maybe just add a test.

Reply:
@lukastaegert Also function type

> Ah, I see `JSON.stringify(undefined)` returns `undefined`. So maybe just add a test.